src/lsm/lsm_stat.c

```diff
@@ -39,7 +39,7 @@ __curstat_lsm_init(
 	WT_ERR(__wt_scr_alloc(session, 0, &uribuf));
 
 	/* Propagate all, fast and/or clear to the cursors we open. */
-	if (!F_ISSET(cst, WT_CONN_STAT_NONE)) {
+	if (cst != 0) {
```

Comment:
I wanted to check if the flags are 0, and copy them only if the value for flags is not 0.
I have fixed that now.